rust/main/agents/relayer/src/server/messages/insert_messages.rs

```diff
@@ -0,0 +1,402 @@
+use std::collections::HashMap;
+
+use axum::{extract::State, http::StatusCode, routing::post, Json, Router};
+use derive_new::new;
+use hyperlane_base::db::{HyperlaneDb, HyperlaneRocksDB};
+use hyperlane_core::{HyperlaneMessage, H256};
+use serde::{Deserialize, Serialize};
+
+use crate::server::utils::{ServerErrorResponse, ServerResult, ServerSuccessResponse};
+
+#[derive(Clone, Debug, new)]
+pub struct ServerState {
+    pub dbs: HashMap<u32, HyperlaneRocksDB>,
+}
+
+impl ServerState {
+    pub fn router(self) -> Router {
+        Router::new()
+            .route("/messages", post(handler))
+            .with_state(self)
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Message {
+    pub version: u8,
+    pub nonce: u32,
+    pub origin: u32,
+    pub sender: H256,
+    pub destination: u32,
+    pub recipient: H256,
+    pub body: Vec<u8>,
+
+    pub dispatched_block_number: u64,
+}
```

Comment:
I wonder if the following may work:

```suggestion
#[derive(Clone, Debug, Deref, Deserialize, Serialize)]
pub struct Message {
    #[deref]
    pub message: HyperlaneMessage
    pub dispatched_block_number: u64,
}
```

Reply:
Updated, needed to add `Deserialize` to `HyperlaneMessage` though